aws/resource_aws_eip_association.go

```diff
@@ -157,7 +157,30 @@ func resourceAwsEipAssociationRead(d *schema.ResourceData, meta interface{}) err
 		return err
 	}
 
-	response, err := conn.DescribeAddresses(request)
+	var response *ec2.DescribeAddressesOutput
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+		response, err = conn.DescribeAddresses(request)
+
+		if tfawserr.ErrCodeEquals(err, "InvalidAssociationID.NotFound") {
```

Comment:
It seems like we should also only retry this error if it is during resource recreation:

```suggestion
		if d.IsNewResource() && tfawserr.ErrCodeEquals(err, "InvalidAssociationID.NotFound") {
```

Reply:
true, updated. Thanks @bflad !